packages/components/src/utils/hooks/stories/use-cx.js

```diff
@@ -93,6 +94,72 @@ export const _slotFillSimple = () => {
 	);
 };
 
+// export const _iframe = () => {
+// 	const iFrameRef = useRef( null );
+
+// 	const redText = css`
+// 		color: red;
+// 	`;
+// 	const blueText = css`
+// 		color: blue;
+// 	`;
+
+// 	return (
+// 		<Iframe ref={ iFrameRef }>
+// 			<Example args={ [ redText ] }>This text should be red</Example>
+// 			<StyleProvider document={ iFrameRef.current?.contentDocument }>
+// 				<Example args={ [ blueText ] }>
+// 					This text should be blue
+// 				</Example>
+// 			</StyleProvider>
+// 		</Iframe>
+// 	);
+// };
+// export const _iframe = () => {
+// 	const redText = css`
+// 		color: red;
+// 	`;
+// 	const blueText = css`
+// 		color: blue;
+// 	`;
+
+// 	const Content = ( { iframeDocument } ) => (
+// 		<>
+// 			<Example args={ [ redText ] }>This text should be red</Example>
+// 			<StyleProvider document={ iframeDocument }>
+// 				<Example args={ [ blueText ] }>
+// 					This text should be blue
+// 				</Example>
+// 			</StyleProvider>
+// 		</>
+// 	);
+
+// 	return (
+// 		<Iframe>
+// 			<Content />
+// 		</Iframe>
+// 	);
+// };
```

Comment:
Solution no. 2: modify the `Iframe` component to pass the `iframeDocument` prop to its children (see changes in `packages/block-editor/src/components/iframe/index.js`)